src/engine/channelmixer_autogen.cpp

```diff
@@ -98,17 +117,18 @@ void ChannelMixer::mixChannels(const EngineMaster::GainCalculator& gainCalculato
         }
         gainCache2.m_gain = newGain[2];
         CSAMPLE* pBuffer2 = pChannel2->m_pBuffer;
-        SampleUtil::copy3WithGain(pOutput,
-                                  pBuffer0, newGain[0],
-                                  pBuffer1, newGain[1],
-                                  pBuffer2, newGain[2],
-                                  iBufferSize);
+        // Process effects for each channel and mix the processed signal into pOutput
+        pEngineEffectsManager->processPostFaderAndMix(pChannel0->m_handle, outputHandle, pBuffer0, pOutput, iBufferSize, iSampleRate, pChannel0->m_features, oldGain[0], newGain[0]);
+        pEngineEffectsManager->processPostFaderAndMix(pChannel1->m_handle, outputHandle, pBuffer1, pOutput, iBufferSize, iSampleRate, pChannel1->m_features, oldGain[1], newGain[1]);
+        pEngineEffectsManager->processPostFaderAndMix(pChannel2->m_handle, outputHandle, pBuffer2, pOutput, iBufferSize, iSampleRate, pChannel2->m_features, oldGain[2], newGain[2]);
```

Comment:
ChannelMixer::applyEffectsInPlaceAndMixChannels still uses an autogenerated loop [at the end](https://github.com/Be-ing/mixxx/blob/82dd24564aa534ecf473194b379ab2dce9becf44/src/engine/channelmixer_autogen.cpp#L7418). Is that optimization worth the large maintenance burden of this Python script autogenerating C++ code? It would be nice to get rid of it. Working with the Python script was a big pain when developing this branch.

Reply:
Ping. What do you think about removing this Python script?

Reply:
Since it is working now, we should keep it. We have IMHO enough other issues to solve.